pkg/domain/infra/abi/images.go

```diff
@@ -380,39 +380,15 @@ func (ir *ImageEngine) Push(ctx context.Context, source string, destination stri
 		signOptions,
 		&dockerRegistryOptions,
 		nil)
+	if err != nil && errors.Cause(err) != storage.ErrImageUnknown {
```

Comment:
I still think that's wrong.  We're guessing here although we could know whether it's manifest list or image index. The `--all` flag is currently ignored as well. 

```golang
if options.All && imageIsAManifestListOrImageIndex {
   return ir.ManifestPush(...)
}
err = newImage.PushImageTo....
```
That would require to wire `--all` into the remote API but I think it's worth it.

The "guessing" may also yield very different error messages. A `podman push nonexistant` may now return something about manifest lists.

Reply:
Or default to `--all` and have an option to push just the bare manifest list.  This stuff isn't intended to be set in stone on the buildah side until we're more sure we've gotten it right.

Reply:
This is not ignored, it is wired into the podman push command.